File: benchmarks/python/numerical_loop.py
import time

def main() -> None:
    n = 100_000_000
    t0 = time.perf_counter()
    total = 0.0
    for i in range(1, n + 1):
        total += 1.0 / i
    elapsed = time.perf_counter() - t0
    ops_per_sec = n / elapsed if elapsed > 0 else 0.0
    print(
        f"BENCH name=numerical_loop time_sec={elapsed:.6f} "
        f"ops={n} ops_per_sec={ops_per_sec:.0f}"
    )
    if total <= 0.0:
        raise SystemExit(1)


if __name__ == "__main__":
    main()

File: benchmarks/zig/numerical_loop.zig
const std = @import("std");

pub fn main() !void {
    const n: i64 = 100_000_000;
    var timer = try std.time.Timer.start();
    var sum: f64 = 0;
    var i: i64 = 1;
    while (i <= n) : (i += 1) {
        sum += 1.0 / @as(f64, @floatFromInt(i));
    }
    const elapsed = @as(f64, @floatFromInt(timer.read())) / 1e9;
    std.debug.print("BENCH name=numerical_loop time_sec={d:.6} ops={d} ops_per_sec={d:.0}\n", .{
        elapsed, n, @as(f64, @floatFromInt(n)) / elapsed,
    });
    if (sum <= 0) std.process.exit(1);
}
